esp-hal/src/rng.rs

```diff
@@ -119,3 +122,118 @@ impl rand_core::RngCore for Rng {
         Ok(())
     }
 }
+
+/// True Random Number Generator (TRNG) driver
+///
+/// The `Trng` struct represents a true random number generator that combines
+/// the randomness from the hardware RNG and an ADC. This struct provides
+/// methods to generate random numbers and fill buffers with random bytes.
+/// Due to pulling the entropy source from the ADC, it uses the associated
+/// regiters, so to use TRNG we need to "occupy" the ADC peripheral.
+///
+/// ```rust, ignore
```

Comment:
we should check this in CI so we should use `no_run` - not `ignore`

Reply:
Done, thanks!